aries_vcx/src/protocols/did_exchange/state_machine/mod.rs

```diff
@@ -110,3 +119,67 @@ impl<I, S> DidExchange<I, S> {
         }
     }
 }
+
+impl<I, S> DidExchange<I, S> {
+    pub fn our_did_doc(&self) -> &DidDocumentSov {
+        &self.our_did_document
+    }
+
+    pub fn their_did_doc(&self) -> &DidDocumentSov {
+        &self.their_did_document
+    }
+
+    pub fn get_endpoint(&self) -> Option<Url> {
+        let did_document = self.their_did_doc();
+        // todo: no unwrap pls
+        let service = did_document.service().first()?;
+        // todo: will get cleaned up after service is de-generified
+        Some(match service {
```

Comment:
This should be performed by `ServiceSov`

Reply:
moved as helper to aries agent, will probably look different after de-generification of did document

Reply:
It will look different, but does your idea include abolishment of `ServiceSov`?